app/services/open_library_service.rb

```diff
@@ -0,0 +1,13 @@
+class OpenLibraryService
+  def self.execute(isbn)
+    include HTTParty
+    base_uri 'https://openlibrary.org/api/books'
+    response = get('/', query: { bibkeys: "ISBN:#{isbn}", jscmd: 'data', format: 'json' })
+
+    if response["ISBN:#{isbn}"].present?
+      response["ISBN:#{isbn}"].slice('title', 'subtitle', 'number_of_pages', 'authors')
+    else
+      Exception.new('Book not Found')
```

Comment:
Are you catching this exception in the controller?

Reply:
I made a refactor and now is catching the record not found rescue in application controller